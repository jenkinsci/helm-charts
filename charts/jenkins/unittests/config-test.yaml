suite: ConfigMap
templates:
  - config.yaml
tests:
  - it: default config
    asserts:
      - isKind:
          of: ConfigMap
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.labels
          value:
            app.kubernetes.io/component: jenkins-controller
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: jenkins
      - equal:
          path: data["apply_config.sh"]
          value: |-
            set -e
            echo "disable Setup Wizard"
            # Prevent Setup Wizard when JCasC is enabled
            echo $JENKINS_VERSION > /var/jenkins_home/jenkins.install.UpgradeWizard.state
            echo $JENKINS_VERSION > /var/jenkins_home/jenkins.install.InstallUtil.lastExecVersion
            echo "download plugins"
            # Install missing plugins
            cp /var/jenkins_config/plugins.txt /var/jenkins_home;
            rm -rf /usr/share/jenkins/ref/plugins/*.lock
            version () { echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; }
            if [ -f "/usr/share/jenkins/jenkins.war" ] && [ -n "$(command -v jenkins-plugin-cli)" 2>/dev/null ] && [ $(version $(jenkins-plugin-cli --version)) -ge $(version "2.1.1") ]; then
              jenkins-plugin-cli --verbose --war "/usr/share/jenkins/jenkins.war" --plugin-file "/var/jenkins_home/plugins.txt" --latest true;
            else
              /usr/local/bin/install-plugins.sh `echo $(cat /var/jenkins_home/plugins.txt)`;
            fi
            echo "copy plugins to shared volume"
            # Copy plugins to shared volume
            yes n | cp -i /usr/share/jenkins/ref/plugins/* /var/jenkins_plugins/;
            echo "finished initialization"
      - matchSnapshot:
          path: data["plugins.txt"]
  - it: no plugins
    set:
      controller.installPlugins: []
    asserts:
      - equal:
          path: data["apply_config.sh"]
          value: |-
            set -e
            echo "disable Setup Wizard"
            # Prevent Setup Wizard when JCasC is enabled
            echo $JENKINS_VERSION > /var/jenkins_home/jenkins.install.UpgradeWizard.state
            echo $JENKINS_VERSION > /var/jenkins_home/jenkins.install.InstallUtil.lastExecVersion
            echo "finished initialization"
      - equal:
          path: data["plugins.txt"]
          value: ""
  - it: additional plugins config
    set:
      controller:
        additionalPlugins:
          - kubernetes-credentials-provider
    asserts:
      - matchSnapshot:
          path: data["plugins.txt"]
  - it: install latest plugins
    set:
      controller.installLatestPlugins: false
    asserts:
      - equal:
          path: data["apply_config.sh"]
          value: |-
            set -e
            echo "disable Setup Wizard"
            # Prevent Setup Wizard when JCasC is enabled
            echo $JENKINS_VERSION > /var/jenkins_home/jenkins.install.UpgradeWizard.state
            echo $JENKINS_VERSION > /var/jenkins_home/jenkins.install.InstallUtil.lastExecVersion
            echo "download plugins"
            # Install missing plugins
            cp /var/jenkins_config/plugins.txt /var/jenkins_home;
            rm -rf /usr/share/jenkins/ref/plugins/*.lock
            version () { echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; }
            if [ -f "/usr/share/jenkins/jenkins.war" ] && [ -n "$(command -v jenkins-plugin-cli)" 2>/dev/null ] && [ $(version $(jenkins-plugin-cli --version)) -ge $(version "2.1.1") ]; then
              jenkins-plugin-cli --verbose --war "/usr/share/jenkins/jenkins.war" --plugin-file "/var/jenkins_home/plugins.txt" --latest false;
            else
              /usr/local/bin/install-plugins.sh `echo $(cat /var/jenkins_home/plugins.txt)`;
            fi
            echo "copy plugins to shared volume"
            # Copy plugins to shared volume
            yes n | cp -i /usr/share/jenkins/ref/plugins/* /var/jenkins_plugins/;
            echo "finished initialization"
  - it: install latest specified plugins
    set:
      controller.installLatestSpecifiedPlugins: true
    asserts:
      - equal:
          path: data["apply_config.sh"]
          value: |-
            set -e
            echo "disable Setup Wizard"
            # Prevent Setup Wizard when JCasC is enabled
            echo $JENKINS_VERSION > /var/jenkins_home/jenkins.install.UpgradeWizard.state
            echo $JENKINS_VERSION > /var/jenkins_home/jenkins.install.InstallUtil.lastExecVersion
            echo "download plugins"
            # Install missing plugins
            cp /var/jenkins_config/plugins.txt /var/jenkins_home;
            rm -rf /usr/share/jenkins/ref/plugins/*.lock
            version () { echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; }
            if [ -f "/usr/share/jenkins/jenkins.war" ] && [ -n "$(command -v jenkins-plugin-cli)" 2>/dev/null ] && [ $(version $(jenkins-plugin-cli --version)) -ge $(version "2.1.1") ]; then
              jenkins-plugin-cli --verbose --war "/usr/share/jenkins/jenkins.war" --plugin-file "/var/jenkins_home/plugins.txt" --latest true --latest-specified;
            else
              /usr/local/bin/install-plugins.sh `echo $(cat /var/jenkins_home/plugins.txt)`;
            fi
            echo "copy plugins to shared volume"
            # Copy plugins to shared volume
            yes n | cp -i /usr/share/jenkins/ref/plugins/* /var/jenkins_plugins/;
            echo "finished initialization"
