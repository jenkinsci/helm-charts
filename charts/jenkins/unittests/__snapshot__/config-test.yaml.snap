additional plugins config:
  1: |
    |-
      kubernetes:4324.vfec199a_33512
      workflow-aggregator:608.v67378e9d3db_1
      git:5.7.0
      configuration-as-code:1947.v7d33fe23569c
      kubernetes-credentials-provider
default config:
  1: |
    |-
      kubernetes:4324.vfec199a_33512
      workflow-aggregator:608.v67378e9d3db_1
      git:5.7.0
      configuration-as-code:1947.v7d33fe23569c
